src/arwave.py

```diff
@@ -1,122 +1,214 @@
+    msgs.info("Using {:s} file for Sky spectrum".format(skyspec_fil))
+    hdu = fits.open(root+'/data/sky_spec/'+skyspec_fil)
+    archive_wave = hdu[0].data
+    archive_flux = hdu[1].data
+    arx_sky = xspectrum1d.XSpectrum1D.from_tuple((archive_wave, archive_flux))
+
+    # Loop on objects
+    flex_dict = dict(polyfit=[], roots=[], shift=[], subpix=[], corr=[],
+                     corr_cen=[], spec_file=skyspec_fil, smooth=[],
+                     arx_spec=[], sky_spec=[])
+
+    for specobj in slf._specobjs[det-1]:  # for convenience
+
+        # Using boxcar
+        if slf._argflag['reduce']['flexure']['spec'] == 'boxcar':
+            sky_wave = specobj.boxcar['wave'].to('AA').value
+            sky_flux = specobj.boxcar['sky']
+        else:
+            msgs.error("Not ready for this flexure method: {}".format(
+                    slf._argflag['reduce']['flexure']))
+
+        # Generate 1D spectrum for object
+        obj_sky = xspectrum1d.XSpectrum1D.from_tuple((sky_wave, sky_flux))
+
+        #Determine the brightest emission lines
+        msgs.warn("If we use Paranal, cut down on wavelength early on")
+        arx_amp, arx_cent, arx_wid, arx_w, arx_satsnd, arx_yprep = aarc.detect_lines(slf, det, msarc=None, censpec=arx_sky.flux.value, MK_SATMASK=False)
+        obj_amp, obj_cent, obj_wid, obj_w, obj_satsnd, obj_yprep = aarc.detect_lines(slf, det, msarc=None, censpec=obj_sky.flux.value, MK_SATMASK=False)
+
+        #Keep only 5 brightest amplitude lines (xxx_keep is array of indices within arx_w of the 5 brightest)
+        arx_keep = np.argsort(arx_amp[arx_w])[-5:]
+        obj_keep = np.argsort(obj_amp[obj_w])[-5:]
+
+        #Calculate wavelength (Angstrom per pixel)
+        arx_disp = (np.amax(arx_sky.wavelength.value)-np.amin(arx_sky.wavelength.value))/arx_sky.wavelength.size
+        obj_disp = (np.amax(obj_sky.wavelength.value)-np.amin(obj_sky.wavelength.value))/obj_sky.wavelength.size
+
+        #Calculate resolution (lambda/delta lambda_FWHM)..maybe don't need this? can just use sigmas
+        arx_res = (arx_sky.wavelength.value[0]+(arx_disp*arx_cent[arx_w][arx_keep]))/(
+            arx_disp*(2*np.sqrt(2*np.log(2)))*arx_wid[arx_w][arx_keep])
+        obj_res = (obj_sky.wavelength.value[0]+(obj_disp*obj_cent[obj_w][obj_keep]))/(
+            obj_disp*(2*np.sqrt(2*np.log(2)))*obj_wid[obj_w][obj_keep])
+        msgs.info("Resolution of Archive={:g} and Observation={:g}".format(
+            np.median(arx_res), np.median(obj_res)))
+
+        #Determine sigma of gaussian for smoothing
+        arx_sig2 = (arx_disp*arx_wid[arx_w][arx_keep])**2.
+        obj_sig2 = (obj_disp*obj_wid[obj_w][obj_keep])**2.
+
+        arx_med_sig2 = np.median(arx_sig2)
+        obj_med_sig2 = np.median(obj_sig2)
+
+        if obj_med_sig2 >= arx_med_sig2:
+            smooth_sig = np.sqrt(obj_med_sig2-arx_med_sig2)  # Ang
+            smooth_sig_pix = smooth_sig / arx_disp
```

Comment:
Hmmm... I see why having a single arx_disp is helpful here. Now might be a good time to take the median of `arx_disp[arx_idx]`, since we are then estimating the disp values at the location of the sky lines, rather than an average over the whole spectrum.